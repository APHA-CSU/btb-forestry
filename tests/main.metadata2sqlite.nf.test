nextflow_process {

    name "Test Process metadata2sqlite"
    script "main.nf"
    process "metadata2sqlite"

    test("Should convert metadata to sqlite") {

        when {
            params {
            }
            process {
                """
                input[0] =  file("${projectDir}/tests/data/filterMeta_exp.csv")
                input[1] =  file("${projectDir}/tests/data/metaTest.csv")
                input[2] =  file("${projectDir}/tests/data/moveTest.csv")
                input[3] =  file("${projectDir}/tests/data/location.csv")
                input[4] =  file("${projectDir}/tests/data/excluded_exp.csv")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
