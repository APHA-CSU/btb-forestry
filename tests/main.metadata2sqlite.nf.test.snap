{
    "Should convert metadata to sqlite": {
        "content": [
            {
                "0": [
                    "viewbovis.db:md5,1a0416eec6ef734ada5958e4d71bce4d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T07:46:10.75951239"
    }
}